### DISCUSSION TITLE
Avoid flooding

### DISCUSSION THREAD

**Discussion by PeterHunt** [windkh/node-red-contrib-telegrambot#388] [General]
Hello Karl-Heinz,

Somtimes, my BOT seems to be hitting the flooding limits, resulting in 429 errors. The number of messages my BOT will be sending is hard to predict. This can be just a few during night time, up to 1000 per minute during rush hours.

Now I just happend to bump into this;

https://grammy.dev/advanced/flood

Which mentions a link towards auto-retry:

https://grammy.dev/plugins/auto-retry

Do you think there is any chance of implementing this auto retry by fetching the 429 and number of seconds and actually retry to send after this time?

Kind regards,

Peter Hunt.

**Comment by windkh** [comment]
Hi Peter. 
I planned to implement a kind of storage queue if the bot is offline. The same one could be used for the error 429. 
Right now I am on vacation. Will do this after summer. Br kh

**Reply by PeterHunt** [reply]
Ah great. Thanks for the quick reply, enjoy your holiday! :)

**Comment by PeterHunt** [comment]
Hello KH,

Did you have any time yet to look into this issue?

Kind regards,
Peter Hunt

**Comment by windkh** [comment]
Hi Peter, no not yet due to many other projects. This topic is something for snowy winter days

**Comment by PeterHunt** [comment]
Hello Karl-Heinz, sorry to bump this subject again.
Is there possibly any progress on this subject or didn't you have time to dig into it yet?

**Reply by windkh** [reply]
Hi Peter, I am really busy with a lot of other projects... will try to find some time. Sry

**Comment by PeterHunt** [comment]
Hello KH, any update yet? This issue is really bugging me. At this point I'm guessing that my BOT from time to time is sending too much messages. Queueing them will not really make any sense for my use case. But I guess I need to figure out some things. Maybe splitting into multiple BOTs or so. I will try to find a solution

**Comment by windkh** [comment]
@PeterHunt Hi Peter, sorry for delay... I did not find the time to fix this yet. Right now I am really busy at work. I will try to do something about the flooding asap

**Comment by PeterHunt** [comment]
Hello, sorry to remind you again. Did you happen to look into this yet, or is it really a challenge to solve this?

**Comment by windkh** [comment]
@PeterHunt Hi Peter, will try to implement this the next days/weeks. Summer is over new and days are getting shorter... hope I will find some time in the evenings.

**Comment by windkh** [comment]
@PeterHunt another hint: please create an issue in future ...

**Comment by PeterHunt** [comment]
Understood, but at first it was mostly a question, not knowing if really needed to be addressed as issue. Could be a limitation at API-side also. But I will raise an issue for this case :)

Ah I see You already did so...

**Comment by windkh** [comment]
@PeterHunt  please follow the discussion here #413

**Comment by windkh** [comment]
solved

**Comment by PeterHunt** [comment]
KH,

Just returned from holiday.
Currently we are currently migrating to a new server.
When all is set, I will give it a try :)

Kind regards,
Peter Hunt.



Op zo 15 mrt 2026 om 11:13 schreef Karl-Heinz Wind ***@***.***

**Comment by windkh** [comment]
@PeterHunt I hope it works…. Let me know
